compiler/tests/testSuite.bal

```diff
@@ -57,15 +68,34 @@ function testCompileEU(string path, string kind) returns file:Error|io:Error? {
     }
 }
 
-function compileErrorLineNumber(CompileError err) returns int? {
-    if err is io:Error {
-        return ();
+function expectedErrorLocation(CompileError err, string path) returns [string, int]|file:Error|io:Error? {
+    string? modulePath = check moduleDir(path);
+    [string, int]? errorLocation = ();
+    if modulePath is string {
+        foreach var md in check file:readDir(modulePath) {
+            if md.dir {
+                foreach var file in check file:readDir(md.absPath) {
+                    errorLocation = check findErrorLine(file.absPath, errorLocation);
+                }
+            }
+        }
     }
-    else {
-        // JBUG #31334 cast needed
-        err:Detail detail = <err:Detail>err.detail();
-        err:LineColumn? lc = detail.location?.startPos;
-        return lc == () ? () : lc[0];
+    errorLocation = check findErrorLine(path, errorLocation);
+    return errorLocation;
+}
+
+function moduleDir(string filePath) returns string|file:Error? {
+    string subModPath = filePath.substring(0, filePath.length()-4) + ".modules";
+    if check file:test(subModPath, file:IS_DIR) {
+        return check file:normalizePath(subModPath, file:CLEAN);
+    }
+}
+
+function findErrorLine(string filePath, any? currentErrorLocation) returns [string, int]|io:Error? {
```

Comment:
`any` is not the right type. I suggest you add a type definition 

type FilenameLine [string, int];

Reply:
fixed